src/sync/sync_session.cpp

```diff
@@ -514,6 +579,43 @@ bool SyncSession::wait_for_upload_completion_blocking()
     return false;
 }
 
+uint64_t SyncSession::register_progress_notifier(std::function<SyncProgressNotifierCallback> notifier,
+                                                 NotifierType direction,
+                                                 bool is_streaming)
+{
+    std::function<void()> invocation;
+    uint64_t token_value = 0;
+    {
+        std::lock_guard<std::mutex> lock(m_progress_notifier_mutex);
+        bool skip_registration = false;
+        token_value = m_progress_notifier_token;
+        m_progress_notifier_token++;
+        auto current_transferrable = (direction == NotifierType::download
+                                      ? m_current_downloadable
+                                      : m_current_uploadable);
+        NotifierPackage package{
+            std::move(notifier),
+            is_streaming,
+            direction,
+            std::move(current_transferrable)
+        };
```

Comment:
This declaration can easily fit on one line. Not sure there's any reason to split it across 6 instead?